packages/core-data/src/actions.js

```diff
@@ -342,6 +355,13 @@ export function* saveEntityRecord(
 		}
 	} catch ( _error ) {
 		error = _error;
+
+		// If we got to the point in the try block where we made an optimistic update,
+		// we need to roll it back here.
+		if ( persistedEntity ) {
+			yield receiveEntityRecords( kind, name, persistedEntity, undefined, true );
+			yield editEntityRecord( kind, name, recordId, record, { undoIgnore: true } );
```

Comment:
This is make the assumption that `record` was part of the `edits` but that's not necessarily the case here since anyone can call ` saveEntityRecord` with any set of values.

I'm thinking a more logical place for all these optimisic updates and rollback is `saveEditedEntityRecord`

Reply:
Maybe that's a good thing, because of how we use it in `editor` to send the latest serialized content?

Reply:
I think it's broken tbh because the API does more that what it's supposed to do. Imagine someone calling that function from a plugin and polluting the edit state?

Isn't it possible to refactor the editor store to rely on `saveEditedEntityRecord` (maybe with an `editEntityRecord` call before)?

Reply:
Most of the data we need is also already in `saveEntityRecord`.

How does this look to you: https://github.com/WordPress/gutenberg/pull/17532/commits/ecf3b843aa2596163ea81f0c1f18d6ddaeefaa9c?

Reply:
It is better I think, if we were to ship this I'd mark this option as experimental as I still think it's not the right place for this behavior entirely.

Reply:
Nevermind, the issue is that we are restoring `record`, we should just restore the previous edit state and we won't have any issues.

Reply:
https://github.com/WordPress/gutenberg/pull/17532/commits/aa967a0ed1ad2bba6da6235dac772d5215badfdc Done!